File: frontend/src/app/core/models/product.ts
export interface Product {
  id: string;
  code: string;
  description: string;
  stock: number;
  createdAtUtc: string;
}

export interface CreateProductRequest {
  code: string;
  description: string;
  stock: number;
}

export type UpdateProductRequest = CreateProductRequest;

export interface FailureModeResponse {
  enabled: boolean;
}

File: services/inventory-service/Models/Product.cs
namespace InventoryService.Models;

public sealed class Product
{
    public Guid Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int Stock { get; set; }
    public DateTime CreatedAtUtc { get; set; }
}
